### DISCUSSION TITLE
Deepseek V2 API support

### DISCUSSION THREAD

**Discussion by ghost** [open-webui/open-webui#3396] [General]
Deepseek v2 is infeasible to run locally and benches better than GPT-4o on every benchmark. Additionally the API is only 20-something cents per 1 million tokens. It's very quickly becoming quite popular for those reasons and it would be nice to be able to use the API in open webui.

**Comment by justinh-rahb** [comment]
Deepseek API is OpenAI compatible, you can already add this, no dev work needed here.

**Comment by shock** [comment]
[Here](https://gist.github.com/shock/d81365117c7f311be8a585a663c11a10) is a working a manifold pipeline for DeepSeek that I simply adapted from the [OpenAI manifold](https://github.com/open-webui/pipelines/blob/main/examples/pipelines/providers/openai_manifold_pipeline.py).  

I couldn't get it to work as a function, but it works as a pipeline running through [pipelines](https://github.com/open-webui/pipelines).

**Reply by swcrazyfan** [reply]
How can I use this? I'm a bit lost.

**Comment by eakkawat** [comment]
@swcrazyfan @pulpocaminante 
I made this for anyone those want to use Deepseek API

https://openwebui.com/f/xgawatt/DeepseekAPI/

**Comment by vladkras** [comment]
I'm running Open WebUI with Docker Desktop so I ended up using
```
docker run -d -p 3000:8080 \
  -e OPENAI_API_BASE_URL="https://api.deepseek.com" \
  -e OPENAI_API_KEY="YOUR_DEEPSEEK_API_KEY" \
  -e OPENAI_API_MODELS="deepseek-chat" \
  -v open-webui:/app/backend/data \
  --name open-webui \
  --restart always \
  ghcr.io/open-webui/open-webui:main
```
I didn't change any other settings manually 

I don't know wether this will affect any other integrations but now I see `deepseek-flash` and `deepseek-v4-pro` along with my local ollama models in the dropdown

UPD:

I found out (not without the help of this newly launched models) that OPENAI_API_MODELS env var is ignored
